recipes/libblst/meta.yaml

```diff
@@ -0,0 +1,129 @@
+package:
+  name: libblst-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/supranational/blst/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 89772cef338e93bc0348ae531462752906e8fa34738e38035308a7931dd2948f
+  patches:
+    - patches/0001-unix-add-libversion.patch  # [unix]
+    - patches/0001-win-simplify-build.sh.patch  # [win]
+
+build:
+  number: 0
+  script_env:
+    - PKG_MAJOR_VERSION={{ so_major_version }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}  # [not win]
+    - {{ compiler('m2w64_c') }}  # [win]
+    - {{ stdlib('c') }}  # [not win]
+    - {{ stdlib('m2w64_c') }}  # [win]
+    - libxcrypt  # [unix]
+    - llvm-tools  # [win]
+    - sed  # [unix]
+
+outputs:
+  - name: libblst
+    files:
+      - lib/libblst.so  # [linux]
+      - lib/libblst.dylib  # [osx]
+      - Library\\lib\\blst.lib  # [win]
+
+      - lib/pkgconfig/blst.pc  # [unix]
+      - Library\\lib\\pkgconfig\\blst.pc  # [win]
+      - lib/pkgconfig/libblst.pc  # [unix]
+      - Library\\lib\\pkgconfig\\libblst.pc  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage('libblst-' ~ so_major_version) }}
+    requirements:
+      build:
+        - {{ stdlib('c') }}  # [not win]
+        - {{ stdlib('m2w64_c') }}  # [win]
+      host:
+        - {{ pin_subpackage('libblst-' ~ so_major_version, exact=True) }}
+        - {{ pin_subpackage('libblst-headers-' ~ so_major_version, exact=True) }}
+      run:
+        - {{ pin_subpackage('libblst-' ~ so_major_version, exact=True) }}
+        - {{ pin_subpackage('libblst-headers-' ~ so_major_version, exact=True) }}
+    test:
+      commands:
+        - test -f ${PREFIX}/include/blst.h  # [unix]
+        - test -f ${PREFIX}/include/blst_aux.h  # [unix]
+        - if not exist %LIBRARY_INC%\\blst.h exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\blst_aux.h exit 1  # [win]
+
+        - test -f ${PREFIX}/lib/libblst.so  # [linux]
+        - test -f ${PREFIX}/lib/libblst.dylib  # [osx]
+        - if not exist %LIBRARY_LIB%\\blst.lib exit 1  # [win]
+
+
+
+  - name: libblst-headers-{{ so_major_version }}
+    files:
+      - include/blst.h  # [unix]
+      - include/blst_aux.h  # [unix]
+      - Library\\include\\blst.h  # [win]
+      - Library\\include\\blst_aux.h  # [win]
+    test:
+      commands:
+        - test -f ${PREFIX}/include/blst.h  # [unix]
+        - test -f ${PREFIX}/include/blst_aux.h  # [unix]
+        - if not exist %LIBRARY_INC%\\blst.h exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\blst_aux.h exit 1  # [win]
+
+
+
+  - name: libblst-{{ so_major_version }}
+    files:
+      - lib/libblst.so.{{ so_major_version }}  # [linux]
+      - lib/libblst.so.{{ version }}  # [linux]
+      - lib/libblst.{{ so_major_version }}.dylib  # [osx]
+      - lib/libblst.{{ version }}.dylib  # [osx]
+      - Library\\bin\\blst-{{ so_major_version }}.dll  # [win]
+      - Library\\lib\\blst-{{ so_major_version }}.lib  # [win]
```

Comment:
This is only needed at compile time and can/should be omitted from the package.
```suggestion
```

Reply:
Just to be sure, blst-[{ so_major_version }} is the name of the 'primary' library. You're saying that the package should work well with just `blst.lib` being a copy of and describing `blst-0.lib/dll`

Reply:
So the -{{so }} is the linkable library, when I install it, I would expect to be able to link with -lblst-{{ so }},  and the compiler would try to find the associated blst-{{ so }}.lib, or will it also find the blst-{{ so }}.dll directly? My uderstanding was that windows needs both, but I know little about windows as my numerous commits can attest

Reply:
Committed for now

Reply:
You need both at build time, but only the DLL is needed at runtime.